src/plugins/lua/elastic.lua

```diff
@@ -239,257 +635,918 @@ local function get_general_metadata(task)
         scan_real)
     scan_real = 0
   end
-
   r.scan_time = scan_real
 
+  local parts = task:get_text_parts()
+  local lang_t = {}
+  if parts then
+    for _, part in ipairs(parts) do
+        local l = part:get_language()
+        if l and not contains(lang_t, l) then
+          table.insert(lang_t, l)
+        end
+    end
+    if table.getn(lang_t) > 0 then
+      r.language = lang_t
+    else
+      r.language = empty
+    end
+    if table.getn(lang_t) == 1 and lang_t[1] == 'en' then
+      r.non_en = false
+    else
+      r.non_en = true
+    end
+  end
+
+  local fuzzy_hashes = task:get_mempool():get_variable('fuzzy_hashes', 'fstrings')
+  if fuzzy_hashes and #fuzzy_hashes > 0 then
+    local l = {}
+    for _, h in ipairs(fuzzy_hashes) do
+      table.insert(l, h)
+    end
+    r.fuzzy_hashes = l
+  else
+    r.fuzzy_hashes = empty
+  end
+
+  r.received_delay = 0
+  if user then -- calculate received_delay only for incoming traffic
+    local recieved_hop = 2
+    local received_headers = task:get_received_headers()
+    if received_headers[recieved_hop] then
+      if received_headers[recieved_hop]['timestamp'] then
+        r.received_delay = math.floor(rspamd_util.get_time()) - received_headers[recieved_hop]['timestamp']
```

Comment:
I'm not sure it's correct as there are timezones in the world.

Reply:
As as saw it counts timezones

Reply:
Double checked - zones is part of timestamp.